moveit_calibration_plugins/handeye_calibration_target/include/moveit/handeye_calibration_target/handeye_target_base.h

```diff
@@ -225,10 +228,19 @@ class HandEyeTargetBase
       return false;
     }
 
-    if (msg->D.size() != CAMERA_DISTORTION_VECTOR_DIMENSION)
+    if (0 == CAMERA_DISTORTION_MODELS_VECTOR_DIMENSIONS.count(msg->distortion_model))
```

Comment:
```suggestion
    if (0 == CAMERA_DISTORTION_MODELS_VECTOR_DIMENSIONS.count(msg->distortion_model))
```
This fails for the field's default value (empty string). I would suggest guessing the distortion model from the number of dimensions and warning the user with e.g. `No distortion model was supplied. Assuming that the model is "X". Specify a distortion model to avoid this message.`

Reply:
I just noticed that this is the CameraInfo message and not a user request. In that case I would say eh

Reply:
A missing or bad intrinsic calibration often doesn't cause things to break obviously, but to just not quite work right. If something is wrong upstream, I think we should just complain, rather than try to "fix" it.

Reply:
Using `std::find` is more idiomatic way of writing the same thing. :thinking: